native-sql-engine/cpp/src/codegen/arrow_compute/ext/array_appender.h

```diff
@@ -428,6 +432,91 @@ class ArrayAppender<DataType, enable_if_decimal<DataType>> : public AppenderBase
   bool has_null_ = false;
 };
 
+template <typename DataType>
+class ArrayAppender<DataType, enable_if_timestamp<DataType>> : public AppenderBase {
+ public:
+  ArrayAppender(arrow::compute::ExecContext* ctx, AppenderType type = left)
+      : ctx_(ctx), type_(type) {
+    std::unique_ptr<arrow::ArrayBuilder> array_builder;
+    arrow::MakeBuilder(ctx_->memory_pool(), arrow::int64(), &array_builder);
```

Comment:
@zhouyuan this is because TimestampType does not contain "arrow::TypeTraits<DataType>::type_singleton()", so it is seperated out.